miner/worker_builder.go

```diff
@@ -413,15 +413,28 @@ func (w *worker) simulateBundle(
 		ethSentToSystem = new(big.Int)
 	)
 
+	currentState := state.Copy()
+
 	for i, tx := range bundle.Txs {
 		state.SetTxContext(tx.Hash(), i+currentTxCount)
 		sysBalanceBefore := state.GetBalance(consensus.SystemAddress)
 
+		prevState := currentState.Copy()
+		prevGasPool := new(core.GasPool).AddGas(gasPool.Gas())
+
 		receipt, err := core.ApplyTransaction(w.chainConfig, w.chain, &w.coinbase, gasPool, state, env.header, tx,
 			&tempGasUsed, *w.chain.GetVMConfig())
 		if err != nil {
 			log.Warn("fail to simulate bundle", "hash", bundle.Hash().String(), "err", err)
 
+			if containsHash(bundle.DroppingTxHashes, receipt.TxHash) {
```

Comment:
it is possible that `receipt` is nil when err happen, use `tx.Hash()` instead

Reply:
fixed